packages/extension-base/src/koni/api/staking/bonding/tao.ts

```diff
@@ -0,0 +1,196 @@
+import { BN, BN_ZERO } from '@polkadot/util';
+
+// interface DelegateInfo {
+//   address: {
+//     name: string,
+//     url: string,
+//     description: string,
+//       signature: string
+//     }
+//   }
+
+// for test
+const testnetDelegate = {
+  // '5GHczYXpzd5xmNwjxWs63hw9DannNBGDp6tG6aPmsqP5WiwM': {
+  //   name: 'Vune',
+  //   url: 'https://fairchild.dev',
+  //   description: 'Vune is a dev at Opentensor and a BSc CS student at UofT.',
+  //   signature: '2a639f931c61abfc3172db594c986c35f1cc8441970582b9c3b1f0506d518a182a2fe570832f02f86014320f1526189917bfbccf7081622652d12e16e9b1768b'
+  // },
+  '5Fjp4r8cvWexkWUVb756LkopTVjmzXHBT4unpDN6SzwmQq8E': {
+    name: 'TaoPolishNode',
+    url: 'https://taonode.io',
+    description: 'This node is a collective effort of the polish community. We are engaged in evangelizing the project, educating and sharing the knowledge.',
+    signature: '1ca20d4e99a48f400dd9cd4aeca8447da6ab1979e480a1dafddfc52e45e215177c7cdde85f5d042d59a5b1169981afa8d1ae28328e2fc5ce57c3d748c8d09d81'
+  },
+  // '5D1rWnzozRX68ZqKTPXZAFWTJ8hHpox283yWFmsdNjxhRCdB': {
+  //   name: 'RoundTable21',
+  //   url: 'https://roundtable21.com',
+  //   description: 'RoundTable21 is an International, multi-disciplinary team of consultants and advisors partnering alongside leading blockchain startups to offer guidance, expertise, investment and hands-on assistance in every aspect of development.',
+  //   signature: '107638b8edde8f918f7faa2cd1f91b454c13094ed5955d6a409f6e0662f8427075516273728a53923839a5428079151ea0844b5f755362364f04735463dff583'
+  // },
+  '5DG4VHT3gKZDEQ3Tx4oVPpejaz64FeDtNPhbAYTLFBmygHUW': {
+    name: 'WaveTensor',
+    url: 'https://twitter.com/wavetensor',
+    description: 'A new Wave is coming, join the AI revolution on top of Bittensor by staking with us.',
+    signature: '5e072b4752ccbdd4ca3298f336284dfdab347dd133850f4d2f9873e7ea59bd2a8f201732842ec79d2bab3abaf133a06b6bd992940389e42d57802c9b8f855889'
+  }
+};
+
+const fetchDelegates = async () => {
+  return new Promise(function (resolve) {
+    fetch('https://raw.githubusercontent.com/opentensor/bittensor-delegates/main/public/delegates.json', { // Todo: check if this is exactly active validator list info or not
+      method: 'GET'
+    }).then((resp) => {
+      resolve(resp.json());
+    }).catch(console.error);
+  });
+};
+
+const fetchStakingInfo = async (): Promise<Array<Record<string, string>>> => {
+  return new Promise(function (resolve) {
+    fetch('https://taostats.io/data.json', {
+      method: 'GET'
+    }).then((resp) => {
+      resolve(resp.json());
+    }).catch(console.error);
+  });
+};
+
+export async function subscribeTaoStakingMetadata (chain: string, substrateApi: _SubstrateApi, callback: (chain: string, rs: ChainStakingMetadata) => void) {
+  // _TODO: optimize case subscribe an API in a subscription -> Rxjs or fetch interval
+  return substrateApi.api.query.subtensorModule.networkImmunityPeriod(async (_iPeriod: Codec) => {
```

Comment:
The data of ChainStakingMetadata below mostly does not change and currently set directly.

Reply:
the type is always "NOMINATED" or dissapear whenever one unstake all. The era is not needed because it doesn't affect any data. The only change is "expectedReturn" and taostats API calculate it each 5 minutes.

Reply:
So, I think subscribe with an interval of 20 minutes is best suitable